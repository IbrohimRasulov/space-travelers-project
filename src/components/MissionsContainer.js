import React from 'react';
import { useSelector } from 'react-redux';
import MissionTable from './SingleMission';

export default function MissionsContainer() {
  const missionsData = useSelector((state) => state.missionsReducer);
  return (
    <table>
      <thead>
        <tr>
          <td>Mission</td>
          <td>Description</td>
          <td>Status</td>
          <td> </td>
        </tr>
      </thead>
      <tbody>
        {missionsData.map((mission) => (
          <MissionTable
            key={mission.id}
            name={mission.name}
            description={mission.description}
            id={mission.id}
            reserved={mission.reserved}
          />
        ))}
      </tbody>
    </table>
  );
}
