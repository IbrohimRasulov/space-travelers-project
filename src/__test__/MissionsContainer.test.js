import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../Redux/configureStore';
import MissionsContainer from '../components/MissionsContainer';

it('testing rendering MissionsContainer component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <MissionsContainer />
      </Router>
    </Provider>,
  ).toJSON;
  expect(tree).toMatchSnapshot();
});
