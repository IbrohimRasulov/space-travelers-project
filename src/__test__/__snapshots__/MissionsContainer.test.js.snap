// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing rendering MissionsContainer component 1`] = `[Function]`;
